src/libs/analysis.js

```diff
@@ -0,0 +1,94 @@
+import _ from 'lodash/fp'
+import { Ajax } from 'src/libs/ajax'
+import { reportError } from 'src/libs/error'
+
+
+const isSet = entityType => _.endsWith('_set', entityType)
+
+/**
+ * Conditions this needs to handle:
+ *
+ * Method config "rootEntityType": sample (or other singular type)
+ *   input:
+ *     1 entity - create submission with a single job
+ *       entityType: "sample"
+ *       entityName: entity name
+ *       expression: ""
+ *     set - create submission with set with expression
+ *       entityType: "sample_set"
+ *       entityName: sample set name
+ *       expression: "samples"
+ *     many entities - create set, then create submission with set with expression
+ *       entityType: "sample_set"
+ *       entityName: sample set name
+ *       expression: "samples"
+ * Method config: sample_set (or other set type)
+ *   input:
+ *     set - create submission with set
+ *       entityType: "sample_set"
+ *       entityName: sample set name
+ *       expression: ""
+ */
+export const launch = async ({
+  workspaceNamespace, workspaceName,
+  config: { namespace: configNamespace, name: configName, rootEntityType },
+  entityType, entityNames, newSetName, expression, useCallCache = true,
+  onCreateSet, onLaunch, onSuccess, onFailure
+}) => {
+  const entityName = _.head(entityNames)
+
+  const workspace = Ajax().Workspaces.workspace(workspaceNamespace, workspaceName)
+  const methodConfig = workspace.methodConfig(configNamespace, configName)
+
+  if (_.isEmpty(entityNames)) {
+    reportError('No entities selected')
+  } else if (isSet(rootEntityType)) {
+    if (_.size(entityNames) > 1) {
+      reportError('Cannot create a submission for multiple entity sets')
+    } else if (entityType !== rootEntityType) {
+      reportError(`Cannot use ${entityType} with method config that needs ${rootEntityType}`)
```

Comment:
There will probably be server-side failures, but only after the submission is created and they may be harder to understand. Also, these conditions _shouldn't_ happen anyway because of the upstream UX. At least half of the reason I included them was to be explicit about what cases this `launch` function doesn't support. The method config and inputs interactions are complicated. A lot of the time I spent on this was understanding what combinations make sense so I wanted to capture the learning to make it easier the next time someone needs to touch this code.